### PR TITLE
CMake platform abstractions

### PR DESCRIPTION
**Pull request by jankrassnigg**
This is a test to see whether these kind of CMake changes would work for all platforms.

Work in progress, final design not clear yet.

### CODE REVIEW

File: Code/BuildSystem/CMake/Platforms/Detect_Linux.cmake [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+		set_property(GLOBAL PROPERTY EZ_CMAKE_PLATFORM_POSTFIX "Linux")
+
+endif()
+
+
+macro(ez_platform_set_build_flags_clang)
+	target_compile_options(${TARGET_NAME} PRIVATE -fPIC)
+
+	# Look for the super fast ld compatible linker called "mold". If present we want to use it.
+	find_program(MOLD_PATH "mold")
```
Comment (Ingrater) [inline Code/BuildSystem/CMake/Platforms/Detect_Linux.cmake]: the mold detection is duplicated between Configure_Linux and Detect_Linux, is this intentional? Could we deduplicate?

File: Code/BuildSystem/CMake/CMakeUtils/ezUtilsCppFlags.cmake [hunk 2]
```diff
@@ -240,6 +189,11 @@ function(ez_set_build_flags_clang TARGET_NAME)
 	else()
 		target_compile_options(${TARGET_NAME} PRIVATE "--system-header-prefix=\"${CMAKE_SOURCE_DIR}/Code/ThirdParty\"")
 	endif()
+
+	if (COMMAND ez_platformhook_set_build_flags_clang)
```
Comment (Ingrater) [inline Code/BuildSystem/CMake/CMakeUtils/ezUtilsCppFlags.cmake:193]: shouldn't we have something similar for gcc and msvc?

**Comment by Ingrater** [comment]
Our CI does not build with clang on windows and does not build with clang on linux. Once you are happy with the design let me know and I can test these cases.

File: Code/BuildSystem/CMake/Platforms/Detect_Linux.cmake (hunk 1 above)
Comment (jankrassnigg) [reply Code/BuildSystem/CMake/Platforms/Detect_Linux.cmake]: That was an accident (unsaved file in VSCode)

File: Code/BuildSystem/CMake/CMakeUtils/ezUtilsCppFlags.cmake (hunk 2 above)
Comment (jankrassnigg) [reply Code/BuildSystem/CMake/CMakeUtils/ezUtilsCppFlags.cmake:193]: Yes, there is still a lot to do.

**Comment by Ingrater** [comment]
I justed tested this with clang on linux, works fine. Clang on windows is covered by clang-tidy so this should be good to go.

**Comment by jankrassnigg** [comment]
I currently have no motivation to continue with this, so going to check this in now.

**Pull request merged**